default.nix

```diff
@@ -1,15 +1,26 @@
 let requiredVersion = import ./lib/minver.nix; in
 
 if ! builtins ? nixVersion || builtins.compareVersions requiredVersion builtins.nixVersion == 1 then
+   let
+    fallback-paths = import ./nixos/modules/installer/tools/nix-fallback-paths.nix;
 
-  abort ''
+   in abort ''
 
     This version of Nixpkgs requires Nix >= ${requiredVersion}, please upgrade:
 
     - If you are running NixOS, `nixos-rebuild' can be used to upgrade your system.
 
-    - Alternatively, with Nix > 2.0 `nix upgrade-nix' can be used to imperatively
-      upgrade Nix. You may use `nix-env --version' to check which version you have.
+    - If you have a multi-user Nix installation on macOS, update Nix by running:
+
+          sudo -i nix-store -r ${fallback-paths.x86_64-darwin};
+          sudo -i ${fallback-paths.x86_64-darwin}/bin/nix-env --uninstall nix
```

Comment:
This shouldn't be necessary given the next step, and might result in the user loosing his reference to nix-store/nix-env.